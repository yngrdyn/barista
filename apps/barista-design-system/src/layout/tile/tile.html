<div
  *ngIf="_hasBadge"
  class="ba-tile-status"
  [class.ba-tile-badge-warning]="_deprecated || _experimental"
  [class.ba-tile-badge-workinprogress]="_workinprogress"
  [class.ba-tile-badge-favorite]="_favorite"
>
  <svg class="ba-tile-status-icon-triangle" viewBox="0 0 50 50">
    <polygon points="50,0 0,0 50,50" />
  </svg>
  <svg
    *ngIf="_workinprogress"
    class="ba-tile-status-icon"
    xmlns="http://www.w3.org/2000/svg"
    width="512"
    height="512"
    viewBox="0 0 512 512"
  >
    <path
      d="M64.199 343.67l58.577-58.577 82.13 22.008 22.005 82.131-58.569 58.569c30.577 1.826 61.757-8.91 85.12-32.272 29.481-29.49 38.894-71.431 28.264-108.889l160.531-160.53c7.657-7.657 7.657-20.07.002-27.727L393.63 69.744c-7.656-7.658-20.071-7.659-27.729-.001L205.368 230.277c-37.458-10.624-79.406-1.212-108.889 28.264-23.365 23.366-34.104 54.556-32.28 85.13z"
    />
    <path
      d="M112.476 399.53l35.707 9.544 26.164-26.161-9.544-35.709-35.709-9.545-26.164 26.162z"
    />
  </svg>
  <svg
    *ngIf="_experimental"
    class="ba-tile-status-icon"
    xmlns="http://www.w3.org/2000/svg"
    width="512"
    height="512"
    viewBox="0 0 512 512"
  >
    <path
      d="M443.9 436.6l-4.2-28.7c-.9-6.3-4.4-15.8-7.7-21.2L311 189.1c-3.3-5.4-6-15-6-21.3V115c0-6.3 5.1-11.5 11.4-11.5h2.1c6.3 0 12.8-5 14.6-11.1l5-17.3c1.8-6.1-1.9-11.1-8.2-11.1H182.1c-6.3 0-9.9 5-8.2 11.1l5 17.3c1.8 6.1 8.3 11.1 14.6 11.1h2.1c6.3 0 11.4 5.2 11.4 11.5v52.8c0 6.3-2.7 15.9-6 21.3L80 386.7c-3.3 5.4-6.7 14.9-7.7 21.2l-4.2 28.7c-.9 6.3 3.4 11.4 9.7 11.4h356.4c6.2 0 10.6-5.1 9.7-11.4zm-71.4-30.1h-233c-4 0-5.6-2.8-3.5-6.3l33.6-55.3c2.1-3.5 7.1-6.3 11.2-6.3H335c4 0 8.9 2.9 10.9 6.5l30.4 55c1.9 3.5.3 6.4-3.8 6.4z"
    />
  </svg>
  <svg
    *ngIf="_favorite"
    class="ba-tile-status-icon"
    xmlns="http://www.w3.org/2000/svg"
    width="512"
    height="512"
    viewBox="0 0 512 512"
  >
    <path
      d="M394.44 468l-52.875-161.952L480 205.956H308.888L255.998 44l-52.872 161.956H32l138.44 100.092L117.57 468l138.427-100.09z"
    />
  </svg>
  <svg
    *ngIf="_deprecated"
    class="ba-tile-status-icon"
    xmlns="http://www.w3.org/2000/svg"
    width="512"
    height="512"
    viewBox="0 0 512 512"
  >
    <path
      d="M256 44L32.224 431.602h447.552L256 44zm28 352h-56v-56.002h56V396zm-56-108V156h56v132h-56z"
    />
  </svg>
</div>
<div class="ba-tile-content">
  <div class="ba-tile-header">
    <div class="ba-tile-img">
      <span>{{ data.identifier }}</span>
    </div>
    <div class="ba-tile-headertext">
      <div class="ba-tile-title">{{ data.title }}</div>
      <div class="ba-tile-type">{{ data.section }}</div>
    </div>
  </div>
  <ng-container *ngIf="!listView">
    <div class="ba-tile-description">
      {{ data.description }}
    </div>
    <div class="ba-tile-footer">
      <span>{{ data.title }} details&nbsp;â†’</span>
    </div>
  </ng-container>
</div>
