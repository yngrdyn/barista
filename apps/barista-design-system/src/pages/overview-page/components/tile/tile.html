<div
  *ngIf="_hasBadge"
  class="ba-tile-status"
  [class.ba-tile-badge-warning]="_deprecated || _experimental"
  [class.ba-tile-badge-workinprogress]="_workinprogress"
  [class.ba-tile-badge-favorite]="_favorite"
>
  <svg class="ba-tile-status-icon-triangle" viewBox="0 0 50 50">
    <polygon points="50,0 0,0 50,50" />
  </svg>
  <img
    *ngIf="_workinprogress"
    src="/assets/maintenance-royalblue.svg"
    alt="Work in progress indicator"
    class="ba-tile-status-icon"
  />
  <img
    *ngIf="_experimental"
    src="/assets/laboratory-white.svg"
    alt="Experimental indicator"
    class="ba-tile-status-icon"
  />
  <img
    *ngIf="_favorite"
    src="/assets/favorite-white.svg"
    alt="Favorite indicator"
    class="ba-tile-status-icon"
  />
  <img
    *ngIf="_deprecated"
    src="/assets/incident-white.svg"
    alt="Deprecated indicator"
    class="ba-tile-status-icon"
  />
</div>
<div class="ba-tile-content">
  <div class="ba-tile-header">
    <div class="ba-tile-identifier">
      {{ data.identifier }}
    </div>
    <div class="ba-tile-headertext">
      <div class="ba-tile-title">{{ data.title }}</div>
      <div class="ba-tile-type">{{ data.section }}</div>
    </div>
  </div>
  <ng-container *ngIf="!listView">
    <div class="ba-tile-description">
      {{ data.description }}
    </div>
    <div class="ba-tile-footer">
      <span>{{ data.title }} details&nbsp;â†’</span>
    </div>
  </ng-container>
</div>
