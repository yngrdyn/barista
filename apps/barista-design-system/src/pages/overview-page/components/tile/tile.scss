@import 'colors';
@import '../../../../styles/global/variables';

$tilesize: 290px;
$tilefontsize: 28px;
$tilefontsize-big: 42px;
$gridgap: 20px;

@mixin tile-theme($link-color, $hover-color, $active-color) {
  color: $link-color;

  :hover,
  &:hover {
    border-color: $hover-color;
    text-decoration: none;
  }

  :focus,
  &:focus {
    outline: 2px solid $hover-color;
  }

  :active,
  &:active {
    border-color: $active-color;
  }

  .ba-tile-img {
    color: $hover-color;
  }

  .ba-tile-badge-favorite,
  &.ba-tile-badge-favorite {
    .ba-tile-status-icon-triangle {
      fill: $link-color;
    }
  }

  .ba-tile-badge-workinprogress,
  &.ba-tile-badge-workinprogress {
    .ba-tile-status-icon-triangle {
      fill: #ffffff;
    }

    svg {
      transform: rotate(90deg);
      fill: $link-color;
    }
  }
}

:host {
  display: block;
  background-color: #ffffff;
  border: 1px solid $gray-200;
  border-radius: 3px;
  height: 100%;
  min-width: $tilesize;
  position: relative;
  padding: 1rem 1.5rem;
  overflow: hidden;

  @include tile-theme($turquoise-600, $turquoise-600, $turquoise-700);

  .ba-tile-badge-warning {
    .ba-tile-status-icon-triangle {
      fill: $yellow-600;
    }
  }

  .ba-tile-content {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100%;
  }

  .ba-tile-header {
    display: flex;
    align-items: center;
  }

  .ba-tile-headertext {
    font-size: 1.2rem;
  }

  .ba-tile-img {
    font-size: $tilefontsize-big;
    margin-right: 1.5rem;
    min-width: 60px;
    text-align: center;

    svg {
      max-width: 50px;
      height: 100%;
      vertical-align: middle;
    }
  }

  .ba-tile-type {
    font-size: 0.9rem;
    color: $gray-500;
    line-height: 1.2;
  }

  .ba-tile-title {
    font-size: 1.4rem;
    line-height: 1.2;
  }

  .ba-tile-description {
    margin: 1rem 0 0;
    flex-grow: 1;
    color: $textcolor;
  }

  .ba-tile-footer {
    color: $gray-500;
    margin-top: 2rem;
  }

  .ba-tile-status {
    position: absolute;
    width: 50px;
    top: 0;
    right: 0;
  }

  .ba-tile-status-icon {
    width: 20px;
    height: 20px;
    position: absolute;
    top: 3px;
    right: 5px;
    fill: #ffffff;
  }
}

.dt-theme-blue :host {
  @include tile-theme($blue-600, $blue-600, $blue-700);
}

.dt-theme-purple :host {
  @include tile-theme($purple-600, $purple-600, $purple-700);
}

.dt-theme-royalblue :host {
  @include tile-theme($royalblue-500, $royalblue-500, $royalblue-700);
}

.dt-theme-turquoise :host {
  @include tile-theme($turquoise-600, $turquoise-600, $turquoise-700);
}
