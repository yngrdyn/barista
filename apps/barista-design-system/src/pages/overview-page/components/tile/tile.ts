/**
 * @license
 * Copyright 2020 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {
  Component,
  ElementRef,
  Input,
  OnChanges,
  SimpleChanges,
} from '@angular/core';

import { BaCategoryNavigationSectionItem } from '@dynatrace/shared/barista-definitions';

@Component({
  selector: 'a[ba-tile]',
  templateUrl: 'tile.html',
  styleUrls: ['tile.scss'],
  host: {
    class: 'ba-tile',
  },
})
export class BaTile implements OnChanges {
  @Input() data: BaCategoryNavigationSectionItem;
  @Input() listView = true;

  /** @internal whether the tile has the badge 'favorite' */
  _favorite = false;

  /** @internal whether the tile has the badge 'deprecated' */
  _deprecated = false;

  /** @internal whether the tile has the badge 'experimental' */
  _experimental = false;

  /** @internal whether the tile has the badge 'workinprogress' */
  _workinprogress = false;

  /** @internal whether the tile has a badge */
  _hasBadge = false;

  constructor(private _elementRef: ElementRef) {}

  /** set the focus on the nativeElement */
  focus(): void {
    this._elementRef.nativeElement.focus();
  }

  ngOnChanges(changes: SimpleChanges): void {
    if (changes.data) {
      this._setBadge();
    }
  }

  /** if the tile has badge properties, only set one of the badge states to true, as only one state can be displayed */
  private _setBadge(): void {
    this._favorite = false;
    this._hasBadge = false;
    this._deprecated = false;
    this._experimental = false;

    if (this.data.badge && this.data.badge.includes('favorite')) {
      this._favorite = true;
    } else if (this.data.badge && this.data.badge.includes('deprecated')) {
      this._deprecated = true;
    } else if (this.data.badge && this.data.badge.includes('experimental')) {
      this._experimental = true;
    } else if (
      this.data.badge &&
      this.data.badge.includes('work in progress')
    ) {
      this._workinprogress = true;
    }

    this._hasBadge =
      this._favorite ||
      this._deprecated ||
      this._experimental ||
      this._workinprogress;
  }
}
