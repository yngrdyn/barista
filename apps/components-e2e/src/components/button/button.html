<section>
  <div>
    <p>
      isDisabled: {{ isDisabled }}, clickCounter:
      <span id="click-counter">{{ clickCounter }}</span>
    </p>
    <button dt-button (click)="isDisabled = !isDisabled" id="disable-toggle">
      Disable buttons
    </button>
  </div>

  <button
    dt-button
    [disabled]="isDisabled"
    (click)="clickCounter = clickCounter + 1"
    id="test-button"
  >
    off
  </button>
</section>
