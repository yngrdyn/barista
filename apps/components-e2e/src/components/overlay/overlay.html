<button id="disable-toggle" (click)="disabled = !disabled">Disable</button>
<span
  id="trigger"
  [dtOverlay]="overlay"
  [dtOverlayConfig]="{ pinnable: true }"
  [disabled]="disabled"
>
  Hover me
</span>
<ng-template #overlay>
  <p>Overlay content</p>
  <button dt-button>Focus me</button>
  <button dt-button>Focus me next</button>
</ng-template>
