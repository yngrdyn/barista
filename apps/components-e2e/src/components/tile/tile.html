<section>
  <div>
    <p>
      isDisabled: {{ isDisabled }}, clickCounter:
      <span id="click-counter">{{ clickCounter }}</span>
    </p>
    <button dt-button (click)="isDisabled = !isDisabled" id="disable-toggle">
      Disable tiles
    </button>
  </div>

  <dt-tile
    [disabled]="isDisabled"
    (click)="clickCounter = clickCounter + 1"
    id="test-tile"
  >
    <dt-tile-title>Tile</dt-tile-title>
    Network traffic
  </dt-tile>
</section>
