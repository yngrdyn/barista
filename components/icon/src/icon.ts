/**
 * @license
 * Copyright 2020 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {
  Attribute,
  ChangeDetectionStrategy,
  Component,
  ElementRef,
  Input,
  OnChanges,
  SimpleChanges,
  ViewEncapsulation,
} from '@angular/core';
import { DtIconType } from '@dynatrace/barista-icons';
import { take } from 'rxjs/operators';

import {
  DtLoggerFactory,
  setComponentColorClasses,
} from '@dynatrace/barista-components/core';

import { DtIconRegistry } from './icon-registry';

export type DtIconColorPalette =
  | 'main'
  | 'accent'
  | 'warning'
  | 'error'
  | 'cta'
  | 'recovered'
  | 'light'
  | 'dark';

const iconLogger = DtLoggerFactory.create('DtIcon');

@Component({
  selector: 'dt-icon',
  exportAs: 'dtIcon',
  template: '<ng-content></ng-content>',
  styleUrls: ['./icon.scss'],
  host: {
    role: 'img',
    class: 'dt-icon',
  },
  changeDetection: ChangeDetectionStrategy.OnPush,
  // Disabled view encapsulation because we need to access and style
  // the dynamically loaded and generated svg elements.
  encapsulation: ViewEncapsulation.None,
})
export class DtIcon implements OnChanges {
  /** Name of the icon in the registry. */
  @Input() name: DtIconType;

  /**
   * Fill color of the svg icon.
   * We can not use the color mixin here because icon has a special extended set of colors.
   */
  @Input()
  get color(): DtIconColorPalette {
    return this._color;
  }
  set color(value: DtIconColorPalette) {
    if (value !== this._color) {
      setComponentColorClasses(this, value);
      this._color = value;
    }
  }
  private _color: DtIconColorPalette;

  constructor(
    public _elementRef: ElementRef,
    private _iconRegistry: DtIconRegistry,
    @Attribute('aria-hidden') ariaHidden: string,
  ) {
    // If the user has not explicitly set aria-hidden, mark the icon as hidden, as this is
    // the right thing to do for the majority of icon use-cases.
    if (!ariaHidden) {
      _elementRef.nativeElement.setAttribute('aria-hidden', 'true');
    }
  }

  ngOnChanges(changes: SimpleChanges): void {
    if (changes.name) {
      if (this.name) {
        this._iconRegistry
          .getNamedSvgIcon(this.name)
          .pipe(take(1))
          .subscribe(
            svg => {
              this._setSvgElement(svg);
            },
            // We do not break the app when an icon could not be loaded
            // so do only a log here
            (err: Error) => {
              iconLogger.warn(
                `Error retrieving icon: ${this.name} ${err.message}`,
              );
            },
          );
      } else {
        this._clearSvgElement();
      }
    }
  }

  private _setSvgElement(svg: SVGElement): void {
    this._clearSvgElement();
    this._elementRef.nativeElement.appendChild(svg);
  }

  private _clearSvgElement(): void {
    const layoutElement: HTMLElement = this._elementRef.nativeElement;
    const childCount = layoutElement.childNodes.length;

    // Remove existing child nodes and add the new SVG element. Note that we can't
    // use innerHTML, because IE will throw if the element has a data binding.
    for (let i = 0; i < childCount; i++) {
      layoutElement.removeChild(layoutElement.childNodes[i]);
    }
  }
}
