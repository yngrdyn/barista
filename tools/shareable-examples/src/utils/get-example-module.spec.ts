/**
 * @license
 * Copyright 2020 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { createTransformer } from './get-example-module';
import {
  createSourceFile,
  ScriptTarget,
  ScriptKind,
  SourceFile,
  transform,
  createPrinter,
} from 'typescript';

function transformFile(
  moduleSource: SourceFile,
  exampleRoot: string,
  exampleClassName: string,
): string {
  const result = transform(moduleSource, [
    createTransformer(exampleRoot, exampleClassName),
  ]);

  // To apply the changes the transformed file needs to be passed
  // to a printer to get the updated file.
  const printer = createPrinter();
  return printer.printFile(result.transformed[0]);
}

it('should remove all relative imports, which are not related to the current example', () => {
  const source = createSourceFile(
    'dummy.ts',
    `
import { NgModule } from '@angular/core';
import { DtOverlayModule } from '@dynatrace/barista-components/overlay';
import { DtKeyValueListModule } from '@dynatrace/barista-components/key-value-list';
import { DtButtonModule } from '@dynatrace/barista-components/button';
import { DtIconModule } from '@dynatrace/barista-components/icon';
import { DtTileModule } from '@dynatrace/barista-components/tile';

import { DtExampleOverlayComplexContent } from './overlay-complex-content-example/overlay-complex-content-example';
import { DtExampleOverlayDefault } from './overlay-default-example/overlay-default-example';
import { DtExampleOverlayImplicitContext } from './overlay-implicit-context-example/overlay-implicit-context-example';

    `,
    ScriptTarget.Latest,
    true, // setParentNodes
    ScriptKind.TS,
  );
  expect(
    transformFile(
      source,
      'barista-components/src/overlay-default-example/overlay-default-example',
      'DtExampleOverlayDefault',
    ),
  ).toMatchSnapshot();
});

it('should remove all relative imports, which are not related to the current example', () => {
  const source = createSourceFile(
    'dummy.ts',
    `
import { NgModule } from '@angular/core';
import { DtOverlayModule } from '@dynatrace/barista-components/overlay';
import { DtKeyValueListModule } from '@dynatrace/barista-components/key-value-list';
import { DtButtonModule } from '@dynatrace/barista-components/button';
import { DtIconModule } from '@dynatrace/barista-components/icon';
import { DtTileModule } from '@dynatrace/barista-components/tile';

import { SomeService } from './example.service.ts';
import { DtExampleOverlayComplexContent } from './overlay-complex-content-example/overlay-complex-content-example';
import { DtExampleOverlayDefault } from './overlay-default-example/overlay-default-example';
import { DtExampleOverlayImplicitContext } from './overlay-implicit-context-example/overlay-implicit-context-example';
    `,
    ScriptTarget.Latest,
    true, // setParentNodes
    ScriptKind.TS,
  );
  expect(
    transformFile(
      source,
      'barista-components/src/overlay-default-example/overlay-default-example',
      'DtExampleOverlayDefault',
    ),
  ).toMatchSnapshot();
});

it('should remove the examples array in the module file', () => {
  const source = createSourceFile(
    'dummy.ts',
    `
export const shouldStayArray = ['hello'];

export const DT_OVERLAY_EXAMPLES = [
  DtExampleOverlayComplexContent,
  DtExampleOverlayDefault,
  DtExampleOverlayImplicitContext,
  DtExampleOverlayProgrammatic,
  DtExampleOverlayTile,
];
    `,
    ScriptTarget.Latest,
    true, // setParentNodes
    ScriptKind.TS,
  );
  expect(
    transformFile(
      source,
      'barista-components/src/overlay-default-example/overlay-default-example',
      'DtExampleOverlayDefault',
    ),
  ).toMatchSnapshot();
});

it('should replace declarations and entryComponents in the module decorator', () => {
  const source = createSourceFile(
    'dummy.ts',
    `
@NgModule({
  imports: [
    DtOverlayModule,
    DtButtonModule,
    DtTileModule,
    DtKeyValueListModule,
    DtIconModule,
  ],
  declarations: [...DT_OVERLAY_EXAMPLES, DtExampleOverlayProgrammaticDummy],
  entryComponents: [...DT_OVERLAY_EXAMPLES, DtExampleOverlayProgrammaticDummy],
})
export class DtOverlayExamplesModule {}
    `,
    ScriptTarget.Latest,
    true, // setParentNodes
    ScriptKind.TS,
  );
  expect(
    transformFile(
      source,
      'barista-components/src/overlay-default-example/overlay-default-example',
      'DtExampleOverlayDefault',
    ),
  ).toMatchSnapshot();
});
