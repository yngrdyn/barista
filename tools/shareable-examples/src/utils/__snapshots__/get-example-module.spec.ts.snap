// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should remove all relative imports, which are not related to the current example 1`] = `
"import { NgModule } from '@angular/core';
import { DtOverlayModule } from '@dynatrace/barista-components/overlay';
import { DtKeyValueListModule } from '@dynatrace/barista-components/key-value-list';
import { DtButtonModule } from '@dynatrace/barista-components/button';
import { DtIconModule } from '@dynatrace/barista-components/icon';
import { DtTileModule } from '@dynatrace/barista-components/tile';
import { DtExampleOverlayDefault } from './overlay-default-example/overlay-default-example';
"
`;

exports[`should remove all relative imports, which are not related to the current example 2`] = `
"import { NgModule } from '@angular/core';
import { DtOverlayModule } from '@dynatrace/barista-components/overlay';
import { DtKeyValueListModule } from '@dynatrace/barista-components/key-value-list';
import { DtButtonModule } from '@dynatrace/barista-components/button';
import { DtIconModule } from '@dynatrace/barista-components/icon';
import { DtTileModule } from '@dynatrace/barista-components/tile';
import { SomeService } from './example.service.ts';
import { DtExampleOverlayDefault } from './overlay-default-example/overlay-default-example';
"
`;

exports[`should remove the examples array in the module file 1`] = `
"export const shouldStayArray = ['hello'];
"
`;

exports[`should replace declarations and entryComponents in the module decorator 1`] = `
"@NgModule({
    imports: [
        DtOverlayModule,
        DtButtonModule,
        DtTileModule,
        DtKeyValueListModule,
        DtIconModule,
    ],
    declarations: [DtExampleOverlayDefault, DtExampleOverlayProgrammaticDummy],
    exports: [DtExampleOverlayDefault, DtExampleOverlayProgrammaticDummy],
})
export class DtOverlayExamplesModule {
}
"
`;
